### PR TITLE
raftstore: gc stale learner automatically

### PR DESCRIPTION
**Pull request by BusyJay**
If learner is isolated before it gets informed by leader that it has been removed, it may never know the fact. This pr fixes this by asking pd for the latest information.

### CODE REVIEW

File: src/raftstore/store/peer.rs [hunk 1]
```diff
@@ -719,9 +720,9 @@ impl Peer {
         if duration >= self.cfg.max_leader_missing_duration.0 {
             // Resets the `leader_missing_time` to avoid sending the same tasks to
             // PD worker continuously during the leader missing timeout.
-            self.leader_missing_time = None;
+            self.leader_missing_time = Some(Instant::now());
```
Comment (siddontang) [inline src/raftstore/store/peer.rs]: seem the comment is stale, so here the same tasks can be sent continuously?

Comment (BusyJay) [reply src/raftstore/store/peer.rs]: No, it will resend the requests only after another max missing time.

**Comment by overvenus** [comment]
Could you fix the `error!` message? It's not a `compact cf-lock tick`.

https://github.com/pingcap/tikv/blob/a109781c26db1315780efc8117e323110ee4597d/src/raftstore/store/store.rs#L2836-L2844

File: src/raftstore/store/peer.rs (hunk 1 above)
Comment (BusyJay) [reply src/raftstore/store/peer.rs]: How? The comment says
>  avoid sending the same tasks to PD worker continuously ***during*** the leader missing timeout.

**Comment by BusyJay** [comment]
PTAL

**Comment by BusyJay** [comment]
@siddontang @hicqu PTAL

**Pull request merged**